### ISSUE TITLE
not maintaining electrumx connection

### ISSUE THREAD

**Issue by strickon** [janoside/btc-rpc-explorer#249]
Not sure if this is a bug or a user error, but I can't et btc-rpc-explorer to maintain the connection to ElectrumX.  I have tried with both tcp and ssl.  With both protocols, it seems to open the connection and immediately drop it.  I have printed the startup debugging info as well and with tcp, it doesn't show the connecting being tried, but with tls it does.  I have validated the ElectrumX works both with nc as well as openssl connecting directly to it.  Is there some environment variable I am not setting ?

I am on a Mac running electrum 1.15.0 and the latest pull from git for btc-rpc-explorer.

Everything else seems to be functioning.

BTCEXP_ADDRESS_API=electrumx
BTCEXP_ELECTRUMX_SERVERS=tls://:electrumx:50002

**Comment by janoside** [comment]
Hmm, it looks like your `.env` format might be off (or there's just a typo in this issue) - as it is here, it looks like you have an extra colon character in your config. The line:

    BTCEXP_ELECTRUMX_SERVERS=tls://:electrumx:50002

Should probably be:

    BTCEXP_ELECTRUMX_SERVERS=tls://electrumx:50002

Does that change anything?

**Comment by strickon** [comment]
> Hmm, it looks like your `.env` format might be off (or there's just a typo in this issue) - as it is here, it looks like you have an extra colon character in your config. The line:
> 
> ```
> BTCEXP_ELECTRUMX_SERVERS=tls://:electrumx:50002
> ```
> 
> Should probably be:
> 
> ```
> BTCEXP_ELECTRUMX_SERVERS=tls://electrumx:50002
> ```
> 
> Does that change anything?

I set it to 

BTCEXP_ELECTRUMX_SERVERS=tls://127.0.0.1:50002

no change.

When the connection is made on startup,

btc-rpc-explorer shows this
electrumClient TLS Connected to  127.0.0.1 50002 +0ms

electrumx reports this

INFO:ElectrumX:[0] SSL 127.0.0.1:49767, 0 total

so it appears that it makes the connection, but the 0 total seems to indicate that it isn't connected.

**Comment by strickon** [comment]
So I was able to get things working.  I had previously did this

1)clone git
2)npm install
3)set up .env
3)npm start

I was able to resolve things by running "npm up" and resolving any discrepancies.

Now I properly get the status message for electrum
  btcexp:electrumx Connected to ElectrumX @ 127.0.0.1:50002 (["ElectrumX 1.15.0","1.4"]) +0ms

and things work

So it seems to have been a dependency issue.

I also tried installing from scratch in a new directory using npm and not git and still had the same errors.

Maybe the instructions need to be updated or something needs to be updated to make sure the right versions of dependencies are pulled in?
